src/components/core/dropdown/dropdown-base/dropdown-base.tsx

```diff
@@ -0,0 +1,654 @@
+                    } else {
+                        dropdown_list_ref.current.scrollTo?.({ top: item_offset_top, behavior: 'smooth' });
+                    }
+                }
+            }
+        }, [active_index]);
+
+        // Shows selected item when the user uncollapsed the list (when item is not in visible area)
+        React.useEffect(() => {
+            if (is_list_visible && dropdown_list_ref.current) {
+                const item_offset_top = dropdown_list_ref.current.itemOffsetTop;
+                dropdown_list_ref.current.scrollTo?.({ top: item_offset_top, behavior: 'smooth' });
+            }
+        }, [is_list_visible]);
+
+        const onClickOutSide = () => {
+            setIsListVisible(false);
+            setIsActive(false);
+            setFilteredItems(list);
+
+            if (!selected_value) {
+                setInputValue('');
+                setActiveIndex(null);
+                onInputChange?.({ target: { value: '' } } as React.ChangeEvent<HTMLInputElement>);
+                onItemSelection?.({ value: '', text: '' });
+            }
+
+            onBlurHandler?.();
+        };
+
+        useOnClickOutside(dropdown_container_ref, onClickOutSide);
+
+        const isListItemWithinView = (item_top: number) => {
+            const list_top = dropdown_list_ref.current?.getBoundingClientRectOfDropdownList?.().top;
+            const list_height = dropdown_list_ref.current?.listClientHeight;
+
+            if (list_top === undefined) return;
+            if (list_height === undefined) return;
+
+            const wrapper_top = Math.floor(list_top);
+            const wrapper_bottom = Math.floor(list_top) + list_height;
+
+            if (item_top >= wrapper_bottom) return false;
+            return item_top > wrapper_top;
+        };
+
+        const getFilteredItems = (text: string, list: TListItem[]) => {
+            return list.filter((item) => typeof item.text === 'string' && item.text.toLowerCase().includes(text));
+        };
+
+        const filterList = (e: React.ChangeEvent<HTMLInputElement>) => {
+            const text = e.target.value.toLowerCase();
+            setInputValue(e.target.value);
+
+            const new_filtered_items = getFilteredItems(text, list);
+            setFilteredItems(new_filtered_items);
+
+            setActiveIndex(null);
+            setSelectedValue('');
+            last_selected_index.current = null;
+
+            onInputChange?.(e);
+        };
+
+        const handleVisibility = () => {
+            setShouldFocusInput(true);
+            setIsActive(true);
+
+            if (typeof onClickDropdownHandler === 'function') {
+                setIsActive(true);
+                onClickDropdownHandler();
+
+                return;
+            }
+
+            setIsListVisible(!is_list_visible);
+        };
+
+        const onKeyPressed = (e: React.KeyboardEvent<HTMLDivElement>) => {
+            e.stopPropagation();
+
+            switch (e.keyCode) {
```

Comment:
```suggestion
            switch (e.code) {
```
See the description in [Comment1]